ci_cd/utils/versions.py

```diff
@@ -599,28 +666,25 @@ def _ignore_semver_rules(
             f"'patch' (you gave {semver_rules['version-update']!r})."
         )
 
-    if "major" in semver_rules["version-update"]:
-        if latest[0] != current[0]:
-            return True
-
-    elif "minor" in semver_rules["version-update"]:
-        if (
-            len(latest) >= 2
+    if (  # pylint: disable=too-many-boolean-expressions
+        ("major" in semver_rules["version-update"] and latest[0] != current[0])
```

Comment:
The nested conditional statements are quite complex. Breaking down this logic into smaller, more descriptive helper functions would increase readability and maintainability.

Reply:
Sure. But I think this is out-of-scope for this PR. What is implemented here is merely a minimization (and simplification) of the code. But I agree that a bit more context could/should be given to this logic.

Reply:
As an extra comment, I'd say this is (sort of) related to #148, at least the motivation behind creating this issue. I can create another issue to follow up simplification of the logic in this file.

Reply:
Done. See #222